src/server/server_family.cc

```diff
@@ -1069,11 +1070,11 @@ static void RunStage(
 // Start saving a single snapshot of a multi-file dfly snapshot.
 // If shard is null, then this is the summary file.
 GenericError DoPartialSave(fs::path full_filename, const dfly::StringVec& scripts,
```

Comment:
You seem to *always* call this functions with the same extension `dfs.tmp` which makes me wonder why `extension` a function parameter and not a variable defined in the body of `DoPartialSave`. That way you will avoid calling it around with the same extension which is error prone